core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -2855,7 +2858,12 @@ public function toggle($intId=null)
 
 		$this->invalidateCacheTags();
 
-		$this->redirect($this->getReferer());
+		Input::setPost('FORM_SUBMIT', $prevSubmit);
+
+		if (!$blnDoNotRedirect)
```

Comment:
```suggestion
		if (!Environment::get('isAjaxRequest'))
```

How about using this instead? It would save us the third method argument.

Reply:
I don't think that would be a good idea.
 1. we have this argument on a few other DC actions with exactly the same behaviour
 2. I would no longer be able to toggle a field when not being an ajax request (e.g. implementing my own toggling behaviour)

Reply:
a possible option would be to not redirect when `$strSubmittedField` is given, but that's kind of some magic behavour to me.